macos/FluentUI/Button/Button.swift

```diff
@@ -139,6 +146,32 @@ open class Button: NSButton {
 			}
 		}
 	}
+	
+	/// the current state of the button
+	private var buttonState: ButtonState? {
+		didSet {
+			guard oldValue != buttonState else {
+				return
+			}
+			updateImageForButtonState()
+		}
+	}
+	
+	/// Dict of images for different button states
+	private var imagesForButtonState = [ButtonState: NSImage]()
+	
+	public func addImage(forButtonState: ButtonState, image: NSImage) {
+		self.imagesForButtonState[forButtonState] = image;
+	}
+	
+	private func updateImageForButtonState() {
+		if imagesForButtonState[self.buttonState!] != nil {
```

Comment:
don't use !, also you should reuse the value right? if let bar = buttonState, let foo = imageforbuttonState[bar] {